test/e2e/scheduler_predicates.go

```diff
@@ -169,6 +169,28 @@ var _ = Describe("SchedulerPredicates [Serial]", func() {
 
 		err = waitForPodsRunningReady(api.NamespaceSystem, systemPodsNo, podReadyBeforeTimeout)
 		Expect(err).NotTo(HaveOccurred())
+
+		for _, node := range nodeList.Items {
+			Logf("\nLogging pods the kubelet thinks is on node %v before test", node.Name)
```

Comment:
suggestion: turn this into a function that can be reused. Given that this PR is for debugging flaky test and is high priority, I'll let you decide if you want to do that.